### PR TITLE
Adds boss music functionality and gives hierophant it's own thematic fight song!

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Adds compatibility to the code to assign music to individual mobs that are played in full length to the player when aggro'd. Music maintains for the entire length, ends, and is repeated if the mob isn't dead. Music is instantly stopped if the mob generating it dies. Other coding avenues were explored around dynamic music generation and allowing for the music to end more naturally, but they were found to be very buggy or impossible to code due to dreammaker having a very........... tense relationship with audio. It also (had to be reminded of this!), lowers the audio of hierophants attacks to make music more audible.

## Why It's Good For The Game

This has the potential to add so much thematic flavor to mobs and events in game, it's great! Represents a big step forward in terms of sound design on tg. Thanks to John Willard for fully coding!
Check out the theme here --> https://youtu.be/mWgv1CHtd4Y

## Changelog

:cl:
add: Adds boss music functionality.
sound: Adds hierophants boss track, tweaks hierophant boss noises.
/:cl:

### CODE REVIEW

**Comment by tommysalami3** [comment]
https://github.com/tgstation/tgstation/assets/77305289/f474e365-6ad6-49a2-b780-dd383437427a

Here's a video of the audio in game.

File: code/datums/looping_sounds/_looping_sound.dm [hunk 1]
```diff
@@ -52,8 +52,9 @@
 	var/loop_started = FALSE
 	/// If we're using cut_mid, this is the list we cut from
 	var/list/cut_list
-	/// The index of the current song we're playing in the mid_sounds list, only used if in_order is used
-	var/audio_index = 1
+	///The index of the current song we're playing in the mid_sounds list, only used if in_order is used
+	///This is immediately set to 1, so we start the index at 0
+	var/audio_index = 0
```
Comment (JohnFulpWillard) [inline code/datums/looping_sounds/_looping_sound.dm:57]: No soundloop in the game uses this but I found this was actually bugged (and previously attempted to use soundloops here) so i fixed it quickly

**Comment by kinneb** [comment]
this is so fucking sick, thank you rick

**Comment by Gaxeer** [comment]
I suppose you should stop music on mob being attacked death

**Comment by optimumtact** [comment]
this song is an entire 1/100th of the rsc size.

**Comment by tommysalami3** [comment]
> this song is an entire 1/100th of the rsc size.
> 
> banger tune though.

i have it about halved in size at 650 testing some compression stuff rn. it's pretty fuzzy but this is like the limit. any further in terms of quality or sample rate and it begins to Sludgify ,,,

i'll push the smaller file in a bit.

**Comment by tommysalami3** [comment]
new file is roughly 46% smaller! heart emoji winky cat face spinning shocked emoji gif

**Comment by Fikou** [comment]
u should port the boss music system from my old pr :)

**Pull request merged**